File: src/clients/java/src/main/java/com/tigerbeetle/ClientClosedException.java
package com.tigerbeetle;

/**
 * ClientClosedException is thrown when the client instance is closed and its resources have been
 * freed.
 **/
public final class ClientClosedException extends RequestException {

    public ClientClosedException() {}

    @Override
    public String getMessage() {
        return toString();
    }

    @Override
    public String toString() {
        return "Client was closed.";
    }
}

File: src/clients/dotnet/TigerBeetle/ClientClosedException.cs
using System;
namespace TigerBeetle;

/// <summary>
/// ClientClosedException is thrown when the client instance is closed and
/// its resources have been freed.
/// </summary>
public sealed class ClientClosedException : RequestException
{
    internal ClientClosedException() { }

    public override string Message => "Client was closed.";
}
